File: analysis/analysis.R
# Welcome to the new project!
# Please mind to maintain the structure of the file and the total project
# as it is

# Initialization - clearing up console and variables
cat("\014")
rm(list = ls())
dev.off(dev.list()["RStudioGD"])

# Libraries - Here you should add your own libraries
# Warning: The following libraries are not used at all, they are just examples
library(stats)
library(readxl)

# Input and output folder
inputfolder <- paste(dirname(getwd()), "/input/analysis/", sep = '')
outputfolder <- paste(dirname(getwd()), "/output/analysis/", sep = '')

# Creating the output direction
# If the output folder already exists, a warning will appear in your console.
dir.create(outputfolder)

############## ANALYSIS ###############
# This is the analysis part
# Feel free to write your own code here

# EXAMPLE - Please read it and then remove the lines 27-42,
# in order fill your own code

# Loading file
fn <- paste(inputfolder, 'example.txt', sep = '')
inputstring <- readChar(fn, file.info(fn)$size)

# Analysis - printing
print(inputstring)

# Saving to output file
outputstring <- 'This is the output string.'
outputfile <-paste(outputfolder, 'output.txt', sep = '')
writeLines(c(outputstring), outputfile)

# GOOD LUCK

File: analysis/analysis.py
# Welcome to the new project!
# Please mind to maintain the structure of this file
# and the total project as it is

# Libraries - You hould add your own libraries here
import pathlib
import os

# Main
if __name__ == "__main__":

    # Getting parent directory
    current_path = pathlib.Path(__file__).parent.absolute()
    parent_path = pathlib.Path(current_path).parent

    # Specifying input and output folder
    inputFolder = str(parent_path) + '/input/analysis/'
    outputFolder = str(parent_path) + '/output/analysis/'

    # Checking if output folder exists, otherwise create it
    if not os.path.exists(outputFolder):
        os.makedirs(outputFolder)
    
    ############## ANALYSIS ###############
    # This is the analysis part
    # Feel free to write your own code here

    # EXAMPLE - Please read it and then remove the lines 28-44,
    # in order fill your own code

    # Reading input file
    inputFile = inputFolder + 'example.txt'
    f = open(inputFile, "r")
    print(f.read())
    f.close()
    
    # Storing data to output folder
    outputString = 'This is the output string.'
    outputFile = outputFolder + 'output.txt'
    f = open(outputFile, 'w')
    f.write(outputString)
    f.close()

    # GOOD LUCK